lib/pbench/server/database/models/metadata.py

```diff
@@ -0,0 +1,88 @@
+import enum
+import datetime
+from pbench.server.database.database import Database
+from sqlalchemy import Column, Integer, DateTime, ForeignKey, Text
+from sqlalchemy.orm import validates
+
+
+class MetadataKeys(enum.Enum):
+    FAVORITE = 1
+    SAVED = 2
+
+
+class Metadata(Database.Base):
+    """ Metadata Model for storing user metadata details """
+
+    # TODO: Think about the better name
+    __tablename__ = "metadata"
```

Comment:
I'm having some trouble playing with SQL in my VM, and I happened to notice that we both used `"metadata"` as the name of a table. I think that, for clarity going forward, we should both rename: me to `"dataset_metadata"` and you to `"user_metadata"` so we avoid confusion. I'm also annoyed that I chose the singular `"dataset"` for my Dataset table because I keep typing `"datasets"`, and I think I'm going to change that, as well.

Reply:
Updated